### PR TITLE
Revert "core/exec-credential: complain louder if inherited credential is missing"

### PR DESCRIPTION
**Pull request by poettering**
Reverts systemd/systemd#32689

I don't get what the point of that PR was? can somebody enlighten me?

### CODE REVIEW

File: man/systemd.exec.xml [hunk 1]
```diff
@@ -3385,9 +3385,6 @@ StandardInputData=V2XigLJyZSBubyBzdHJhbmdlcnMgdG8gbG92ZQpZb3Uga25vdyB0aGUgcnVsZX
         a terse way to declare credentials to inherit from the service manager into a service. This option
         may be used multiple times, each time defining an additional credential to pass to the unit.</para>
 
-        <para>Note that if the path is not specified or a valid credential identifier is given, i.e.
```
Comment (YHNdnzj) [inline man/systemd.exec.xml:3388]: At least this should really be kept?

**Review by YHNdnzj — COMMENTED** [review]
Hmm, my idea is like this: `LoadCredential=` should and should only be used when people are sure what they want. When a path is also specified, the missing cred is considered fatal, otherwise a warning is emitted. OTOH, if people want generic/completely optional stuff, they're better off using `ImportCredential=`. Our units are actually doing great, by always using `ImportCredential=` for acquiring credentials passed to service manager.

We could also just make `LoadCredential=` mandatory. However, that would become a complete compat break, which is something I'd like to avoid.

**Comment by YHNdnzj** [comment]
IOW, I wouldn't think a warning makes much sense if we didn't have `ImportCredential=`. But we do, and thus the semantics of `LoadCredential=` and that can have a clear separation.

**Comment by bluca** [comment]
What about just downgrading to info level?

File: src/core/exec-credential.c [hunk 2]
```diff
@@ -443,7 +443,7 @@ static int load_credential(
 
                 /* Pass some minimal info about the unit and the credential name we are looking to acquire
                  * via the source socket address in case we read off an AF_UNIX socket. */
-                if (asprintf(&bindname, "@%" PRIx64 "/unit/%s/%s", random_u64(), unit, id) < 0)
+                if (asprintf(&bindname, "@%" PRIx64"/unit/%s/%s", random_u64(), unit, id) < 0)
```
Comment (keszybz) [inline src/core/exec-credential.c:446]: This also. It is strange to have asymmetrical whitespace.

**Comment by keszybz** [comment]
Let's continue the discussion in https://github.com/systemd/systemd/pull/32815. This revert also reverts the cleanups, which we don't want.

**Pull request closed without merging**